### PR TITLE
Exact should default to `true` for `/` path on router-link

### PR DESCRIPTION
**Pull request by Akryum**
[Rendered](https://github.com/Akryum/rfcs/blob/home-exact/active-rfcs/0000-home-exact.md)

### CODE REVIEW

**Comment by dominik-bln** [comment]
What would happen on something like `<RouterLink
    :to="{name:'home'}">`?

What would happen if the application is hosted at `/my-app/`?

**Comment by Akryum** [comment]
@dominik-bln The resolved path in vue-router would still be `/`.

**Comment by ycmjason** [comment]
I personally prefer things with less exceptional cases and make things more explicit. I agree that `/` should always be `exact` but adding this exceptional case could be confusing for developers.

Instead, I think the correct way to solve this problem is to have all `router-link` to be `exact: true` by default and let user to specify something like `not-exact`. (need to think of a better name for that attr)

**Comment by posva** [comment]
> An alternative way to solve this problem is to have all router-link to be exact: true by default and let developer specify something like not-exact. (need to think of a better name for that attr) I think it is safer to assume routes to always be exact than assuming that each route would have some sub-paths which I wish to add active to.

I also thought about that but the most common use case for active links is navbars and drawers, sometimes with submenus. In those scenarios, you want the active class to be applied while in a subroute. The only case that does not work by default is the `/`. While it is an exception, it isn't something people would have to think about because a `/` that isn't exact is **always** active, it's pretty much the same as having a custom class that is always applied, which seems way more intuitive than relying on the active class to add css

In other words, we want to make apply exact for `/` because that's what we always want, so in most scenarios, people won't even need to use `exact` at all

**Comment by blake-newman** [comment]
I personally agree with @ycmjason 

Adding a different behaviour for a specific use case will lead to more confusion, although there is often a use case that `/` should have exact, it may be not the case. To reverse the default behaviour there will be two input structures:

`<router-link to="/" :exact="false">`
`<router-link to="/path" exact`>`

Without reading the documentation it will not be immediately obvious why there is two different behaviours, although reading the documentation is probably advised if you see unintended behaviour the API should be as intuitive as possible to not require the need for reading the documentation. Thus having the default the same for both cases is preferred IMO. 

I would argue that the default value for exact should be true, however that seems like a subjective default value. It is an assumption that we know the majority use case, and either default will likely result in someone not being happy.

**Comment by posva** [comment]
> I would argue that the default value for exact should be true, however that seems like a subjective default value. It is an assumption that we know the majority use case, and either default will likely result in someone not being happy.

Why would you have `exact` default to true though?

the idea here is really to make things less confusing by eliminating the need to use `exact` except for advanced users, which would have to search the docs for it

**Comment by blake-newman** [comment]
I believe that there is no perfect default in this case. 

However, based on my expectations on first usage I perceived that it would be more intuitive to match the exact path provided, as i would expect to have to provide a prop to allow nested routes to match.

Thus being subjective, as I believe every developer will have separate opinions on what the default behaviour should be. Personally, i would prefer to switch on nested route matching. 

I also believe that this proposal is biased to the use case of a `<nav>` (although an often use case). The use of `<router-link>` in the `<nav>` use case make sense for nested route matching by default. However the opposite can be said in use cases where `<router-link>` is used in places such as content. In this use case _exact_ matching would be preferred, thus making the default value subjective to how the router link is used.

**Comment by ycmjason** [comment]
> I also thought about that but the most common use case for active links is navbars and drawers, sometimes with submenus. In those scenarios, you want the active class to be applied while in a subroute. 

I agree that in navbars/drawers, it is common that we wish to match subroutes. However, it is almost the only place I can think of where we would match subroutes, I would use `exact` in almost every other scenario.

**Comment by posva** [comment]
> However, based on my expectations on first usage I perceived that it would be more intuitive to match the exact path provided, as i would expect to have to provide a prop to allow nested routes to match.

To be clear, there are two classes applied by default, one for active and one for exact

> is used in places such as content. In this use case exact matching would be preferred, thus making the default value subjective to how the router link is used.

Why would you mark links in content with custom style when active? That is much more specific use-case than the navbar and drawer though

> I would use exact in almost every other scenario I can think of. (e.g. breadcrumbs, body etc.)

breadcrumbs is actually another use case of the exact=false, what is the _body_ scenario? I wouldn't mind seeing other scenarios hehe

**Comment by ycmjason** [comment]
> breadcrumbs is actually another use case of the exact=false

Why would you want exact=false in breadcrumbs?

Imagine we have a route `/orgs/vue/members/posva`, the breadcrumbs might show as:

```
Orgs > Vue > Members > Posva
```

In this case, I wouldn't expect `Members` to have the active class.

**Comment by posva** [comment]
I see what you mean now, I was thinking of a deeper breadcrumb, showing further entries that are not active, like being in _Members_ while showing all the way til _Posva_
For that case, you would use the exact-active class instead though

**Comment by Akryum** [comment]
Remember this proposal only concerns the router-link-active class, which is
broken by default for `/`. Currently, the exact prop is to work around this
which is something the new comer has to learn about because router-links
don't behave as expected in his menu.

**Comment by posva** [comment]
This is closed in favour of https://github.com/vuejs/rfcs/pull/136
